### PR TITLE
Use Interface to Allow Custom Binding of HitsIteratorAggregate

### PR DESCRIPTION
**Pull request by ganicus**
Added an interface "HitsIteratorAggregate" that is used in the elastic scout engine's map method when parsing elasticsearch results. 

```php
 public function map(BaseBuilder $builder, $results, $model)
  {

        $hits = app()->makeWith(HitsIteratorAggregate::class, 
                    ['results' => $results, 
                    'callback' => $builder->queryCallback
                    ]);

        return new Collection($hits);
  }
```

It would allow users to define their own implementation by simply overriding the binding in the AppServiceProvider's register method. 

```php
public function register()
{

    $this->app->bind(HitsIteratorAggregate::class, CustomHitsIteratorAggregate::class);

}
```

It currently defaults to the packages EloquentHitsIteratorAggregate in the ElasticSearchServiceProvider class.

```php
$this->app->bind(
     'Matchish\ScoutElasticSearch\ElasticSearch\HitsIteratorAggregate',
     'Matchish\ScoutElasticSearch\ElasticSearch\EloquentHitsIteratorAggregate'
);
```

Currently tested on elasticsearch-7 branch.

### CODE REVIEW

**Comment by ganicus** [comment]
I've conformed to the styleci requirements and I'm getting identical diff issues.

Even when I download and git apply the raw diff file it doesn't change anything in my forked repo. 

I'm also getting sporadic errors from the TravisCI builder with regards to elasticsearch during build.

```bash
The command "curl -X GET "localhost:9200/_cluster/health?wait_for_status=yellow&timeout=50s"" failed and exited with 7 during .
```

Please let me know if you see any issues on your end. Thank you.

**Comment by matchish** [comment]
Great job! Thanks for the help! Just ignore StyleCi failed checks. I'll apply their fix later on my own.
Could you please squash commits? I mean there are commits like [this](https://github.com/matchish/laravel-scout-elasticsearch/pull/49/commits/880931b61daac57a108d5e9f0f1ee8cccf1d7972) that pollute git history. Also, could you update [unreleased](https://github.com/matchish/laravel-scout-elasticsearch/blob/elasticsearch-7/CHANGELOG.md#unreleased) section in `CHANGELOG.md`?

**Comment by ganicus** [comment]
Made the changes specified. Sorry, I've been working on a deadline with another project. Please let me know if everything is satisfactory. Thank you.

File: src/Engines/ElasticSearchEngine.php [hunk 1]
```diff
@@ -117,7 +120,8 @@ public function getTotalCount($results)
 
     /**
      * @param BaseBuilder $builder
-     * @param array $options
+     * @param array       $options
```
Comment (matchish) [inline src/Engines/ElasticSearchEngine.php]: This is unnecessary changes

Comment (ganicus) [reply src/Engines/ElasticSearchEngine.php]: I'll fix this and would "Add HitsIteratorAggregate for custom binding" be a more imperative tone?

Comment (matchish) [reply src/Engines/ElasticSearchEngine.php]: You right it would be more imperative, or you can use `Replace harcoded EloquentHitsIteratorAggregate to an interface`

Comment (ganicus) [reply src/Engines/ElasticSearchEngine.php]: Done, fixed the above mentioned issue and renamed commit to "Replace harcoded EloquentHitsIteratorAggregate to an interface"

**Comment by matchish** [comment]
Thank a lot for the help)

**Comment by ganicus** [comment]
Glad I could help! Thanks for the feedback!
